In [20]:
# Imports
import yfinance as yf
import pandas as pd
import numpy as np

# Part 1: Data Collection and Preparation

In [1]:
tickers = ["AAPL", "NVDA", "AMZN", "GOOGL", "MSFT"]
start = "2022-01-01"
end = "2026-05-01"
interval = "1d"

In [13]:
# Data downloading and saving
# Close price is used because it is the most widely used in literature.
data = yf.download(tickers=tickers, start=start, end=end, interval=interval, auto_adjust=True)["Close"] # adjusted for splits and dividents
data.head(5)
data.to_csv("data/close_prices_full.csv")

In [15]:
data.info()

<class 'pandas.DataFrame'>
DatetimeIndex: 1003 entries, 2022-01-03 to 2025-12-31
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   AAPL    1003 non-null   float64
 1   AMZN    1003 non-null   float64
 2   GOOGL   1003 non-null   float64
 3   MSFT    1003 non-null   float64
 4   NVDA    1003 non-null   float64
dtypes: float64(5)
memory usage: 47.0 KB


After downloading data, the dataset contains 1003 trading days with no missing values

In [24]:
daily_returns = (data - data.shift(1)) /  data.shift(1)
daily_returns = daily_returns.dropna()

log_daily_returns = np.log(data / data.shift(1))
log_daily_returns = log_daily_returns.dropna()

# Save to csv for later usage
daily_returns.to_csv("data/daily_returns_full.csv")
log_daily_returns.to_csv("data/log_daily_returns_full.csv")

# Part 2: Descriptive Statistics